src/app/public/modules/sky-tree-view/sky-tree-view-node-wrapper.component.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Input,
+  OnInit,
+  Optional,
+  ViewEncapsulation
+} from '@angular/core';
+
+import {
+  TREE_ACTIONS,
+  TreeModel,
+  TreeNode
+} from 'angular-tree-component';
+
+import {
+  SkyCheckboxChange
+} from '@skyux/forms';
+
+import {
+  SkyTreeViewComponent
+} from './sky-tree-view.component';
+
+@Component({
+  selector: 'sky-tree-view-node-wrapper',
+  templateUrl: './sky-tree-view-node-wrapper.component.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyTreeViewNodeWrapperComponent implements OnInit {
+
+  @Input()
+  public index: number;
+
+  @Input()
+  public node: TreeNode;
+
+  public set isPartiallySelected(value: boolean) {
+    if (value !== this._isPartiallySelected) {
+      this._isPartiallySelected = value;
+      this.changeDetectorRef.markForCheck();
+    }
+  }
+
+  public get isPartiallySelected(): boolean {
+    return this._isPartiallySelected;
+  }
+
+  public set isSelected(value: boolean) {
+    if (value !== this._isSelected) {
+      this._isSelected = value;
+      this.changeDetectorRef.markForCheck();
+    }
+  }
+
+  public get isSelected(): boolean {
+    return this._isSelected;
+  }
+
+  private _isPartiallySelected: boolean;
+
+  private _isSelected: boolean;
+
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    @Optional() private skyTreeView: SkyTreeViewComponent
```

Comment:
I would throw an error if `this.skyTreeView` is `undefined`.

Reply:
Done.